README.md

```diff
@@ -28,4 +28,294 @@ One of the very common use cases of WebXR is 180 and/or 360 photos or videos, bo
 
 Another example where composition layers are going to shine is displaying text or high resolution textures in the virtual world. Since composition layers allow to sample source texture at full resolution without multiple re-samplings the readability of the text is going to be significantly improved.
 
+## Usage
 
+Using the Layers API consists of three primary steps:
+
+  1. Creating, positioning, and sizing the layers to be shown.
+  1. Creating graphics API resources to back the layers.
+  1. Rendering to the graphics API resources during the frame loop.
+
+### Layer creation
+
+Layers are created by the `XRSession`, using the `requestLayer()` series of fuctions, patterned after the `requestReferenceSpace()` method. Unlike other `request...` methods this function does not return a `Promise`, since it's expected that layer construction will be fairly lightweight. It may return `null`, however, if the `XRSession` does not support the requested layer type. The only layer type that _must_ be supported is `"projection"`. The returned instance depends on the requested type, with each layer type corresponding to it's own interfaces which extends `XRLayer`.
+
+```js
+// Returns an XRProjectionLayer instance
+const projectionLayer = xrSesssion.requestLayer('projection');
+```
+
+This explainer will not cover the details of every possible layer. For additional details please see the separate explainers for each layer type:
+
+  - XRProjectionLayer
+  - [XRQuadLayer](./quad.md)
+  - [XRCylinderLayer](./cylinder.md)
+  - [XREquirectLayer](./equirect.md)
+  - [XRCubeLayer](./cubemap.md)
+
+Layers are not presented to the XR device until they have been added to the `layers` `XRRenderState` property with the `updateRenderState()` method. Setting the `layers` array will override the `baseLayer` if one is present, with `baseLayer` reporting the first element in the `layers` array. Layers will be presented in the order they are given in the `layers` array, with each layers being given in "back-to-front" order. Layers may have alpha blending applied if the layer's `blendSourceAlpha` attribute is `true`, but no depth testing may be performed between layers.
+
+```js
+const projectionLayer = xrSesssion.requestLayer('projection');
+const quadLayer = xrSesssion.requestLayer('quad');
+
+xrSession.updateRenderState({ layers: [projectionLayer, quadLayer] });
+```
+
+`updateRenderState()` _may_ throw an exception if more layers are specified than the `XRSession` supports simultaneously.
+
+### Graphics API binding
```

Comment:
Is this really needed?

Reply:
It is a factory for layer sources and it makes sense (to me).

Reply:
Why not have it on session? 
On the web platform, decorator classes are typically not used.

Reply:
I don't know, actually, I think either way is fine for me as long as we keep the graphics API <-> layer separation. If we put it in session, then the session class becomes kinda bloated and the XRSession now must be aware of all of those Graphics APIs.... This is what @toji are trying to avoid, I assume.